src/archetype/createProject/SelectParentPomStep.ts

```diff
@@ -0,0 +1,68 @@
+import { pathExistsSync } from "fs-extra";
+import { Disposable, QuickInputButtons, QuickPick, QuickPickItem, window } from "vscode";
+import { MavenProjectManager } from "../../project/MavenProjectManager";
+import { IProjectCreationMetadata, IProjectCreationStep, StepResult } from "./types";
+import { MavenProject } from "../../explorer/model/MavenProject";
+
+interface ParentPomPickItem extends QuickPickItem {
+    parentProject?: MavenProject;
+}
+
+export class SelectParentPom implements IProjectCreationStep {
+    previousStep?: IProjectCreationStep | undefined;
+
+    async run(metadata: IProjectCreationMetadata): Promise<StepResult> {
+        const items: ParentPomPickItem[] = [
+            {
+                label: "<None>",
+                parentProject: undefined
+            }
+        ];
+        MavenProjectManager.projects
+            .filter(project => project.artifactId && project.pomPath && pathExistsSync(project.pomPath))
+            .sort((a, b) => a.pomPath.length - b.pomPath.length)
+            .forEach(project => {
+                items.push({
+                    label: project.artifactId,
+                    description: project.pomPath,
+                    parentProject: project,
+                });
+            });
+
+        const disposables: Disposable[] = [];
+        try {
+            return await new Promise<StepResult>((resolve) => {
+                const pickBox: QuickPick<ParentPomPickItem> = window.createQuickPick<ParentPomPickItem>();
+                pickBox.title = metadata.title;
+                pickBox.placeholder = "Select the parent...";
+                pickBox.matchOnDescription = true;
+                pickBox.ignoreFocusOut = true;
+                pickBox.items = items;
```

Comment:
Will it be better if we skip this step when there is only one available item?

Reply:
We always add a `<None>` in the pick items. If only `<None>` is left in the items, I prefer to keep it to make user know there is no parent pom in current workspace.

Reply:
ok